### ISSUE TITLE
Implement clip grad value on FSDP

### ISSUE THREAD

**Issue by piEsposito** [pytorch/pytorch#112171]
### 🚀 The feature, motivation and pitch

Gradient clipping is a popular technique to scale gradients and avoid exploding gradients issues in RNNs/very deep networks. PyTorch offers a util [torch.nn.utils.clip_grad_value_](https://pytorch.org/docs/stable/generated/torch.nn.utils.clip_grad_value_.html) that enables users to clip gradients such that they collectively have a capped maximum value.

While the torch.nn.utils.clip_grad_value_ should work, it is not ideal to have different apis (torch.nn.utils.clip_grad_value_ vs self.clip_grad_norm_) when handling the same feature. In that sense, adding clip_grad_value_ could contribute to a clearer

Call self._lazy_init to ensure stream synchronization is done appropriately
Apply inplace clamp on all gradients
Ensure that this works with nested FSDP instances as well.

That would also help for clearer design over distributed training tools built over PyTorch.

### Alternatives

Use clip_grad_value_ from torch.nn.utils.clip_grad_value_ and self.clip_grad_norm_ when handling gradient norms on FSDP.

cc @zhaojuanmao @mrshenli @rohan-varma @awgu @fegin @penguinwu

**Comment by mori360** [comment]
Gradient clipping is available [here](https://github.com/pytorch/pytorch/blob/5cc1b54a91e239cefab9a1e30a175853b1f07514/docs/source/notes/amp_examples.rst#gradient-clipping).
If you need further support with FSDP2, please open a new issue with updated information.
